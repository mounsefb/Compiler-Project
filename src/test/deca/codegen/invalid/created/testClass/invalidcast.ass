	TSTO #11
	BOV stack_overflow_error
	ADDSP #24
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 1(GB)
	LOAD code.object.equals, R0
	STORE R0, 2(GB)
; code de la table des méthodes de A
	LEA 1(GB), R0
	STORE R0, 3(GB)
	LOAD code.object.equals, R0
	STORE R0, 4(GB)
; code de la table des méthodes de B
	LEA 3(GB), R0
	STORE R0, 5(GB)
	LOAD code.object.equals, R0
	STORE R0, 6(GB)
; start main program
; Main program
; Beginning of main instructions:
	NEW #1, R2
	BOV tas_plein
	LEA 3(GB), R0
	STORE R0, 0(R2)
	PUSH R2
	BSR init.A
	POP R2
	STORE R2, 7(GB)
	LOAD 7(GB), R2
	BRA cast_incorrect
	STORE R2, 8(GB)
	HALT
; end main program
overflow_error:
	WSTR "Error: Overflow during arithmetic operation"
	WNL
	ERROR
stack_overflow_error:
	WSTR "Error: Stack Overflow "
	WNL
	ERROR
io_error:
	WSTR "Error: Input/Output error"
	WNL
	ERROR
Dereferencement.null:
	WSTR "Erreur : dereferencement de null"
	WNL
	ERROR
pile_pleine:
	WSTR "Erreur : pile pleine"
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : allocation impossible, tas plein"
	WNL
	ERROR
cast_incorrect:
	WSTR "la conversion (cast) est incorrecte"
	WNL
	ERROR
; code de l'initialisation des champs
init.A:
	RTS
init.B:
	LOAD -2(LB), R1
	PUSH R1
	BSR init.A
	SUBSP #1
	RTS
; code des méthodes
code.object.equals:
	LOAD -2(LB), R0
	CMP -3(LB), R0
	BEQ if_equal_pour_object.equals
	LOAD #0, R0
	BRA fin_methodes
if_equal_pour_object.equals:
	LOAD #1, R0
	BRA fin_methodes
fin_methodes:
	RTS
