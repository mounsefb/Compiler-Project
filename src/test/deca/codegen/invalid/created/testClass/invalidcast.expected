la conversion (cast) est incorrecte
